apps/teams-test-app/src/components/AppInitialization.tsx

```diff
@@ -2,22 +2,55 @@ import { app } from '@microsoft/teams-js';
 import React, { ReactElement } from 'react';
 
 import BoxAndButton from './BoxAndButton';
+import { ApiWithoutInput, ApiWithTextInput } from './utils';
 
-const AppInitializationAPIs = (): ReactElement => {
-  const [notifyLoadedRes, setNotifyLoadedRes] = React.useState('');
-  const [notifySuccessRes, setNotifySuccessRes] = React.useState('');
-  const [notifyFailureRes, setNotifyFailureRes] = React.useState('');
+const NotifyLoaded = (): React.ReactElement =>
+  ApiWithoutInput({
+    name: 'appInitializationAppLoaded',
+    title: 'appInitialization.appLoaded',
+    onClick: async () => {
+      app.notifyAppLoaded();
+      return 'called';
+    },
+  });
 
-  const notifyLoaded = (): void => {
-    app.notifyAppLoaded();
-    setNotifyLoadedRes('called');
-  };
+const NotifySuccess = (): React.ReactElement =>
+  ApiWithoutInput({
+    name: 'appInitializationSuccess',
+    title: 'appInitialization.success',
+    onClick: async () => {
+      app.notifySuccess();
+      return 'called';
+    },
+  });
 
-  const notifySuccess = (): void => {
-    app.notifySuccess();
-    setNotifySuccessRes('called');
-  };
+const NotifyFailure = (): React.ReactElement =>
+  ApiWithTextInput<app.FailedReason>({
+    name: 'appInitializationFailure2',
```

Comment:
>2 [](http://example.com/codeflow?start=35&length=1)

what happened to 1

Reply:
There isn't "1" per se, we do have `appInitializationFailure` already, which has to stay for now before I can move the E2E scenario tests over to use the new one. So I went with `2` for these new items. Do you have a better suggestion on how to do these N/N-1 cases where tests have to be updated as well?

Reply:
Oh! There is it is. No, I think 2 is fine. Just missed the BoxAndButton at the bottom